src/core/global.c

```diff
@@ -1465,3 +1483,69 @@ void nn_global_rele_socket(struct nn_sock *sock)
     nn_sock_rele(sock);
     nn_glock_unlock();
 }
+
+int nn_setopt (int option, const void *optval, size_t optvallen)
+{
+    int idx;
+
+    switch (option) {
+    case NN_FORK_STRATEGY:
+        nn_assert(optvallen == sizeof (int));
+        idx = *(const int *) optval;
+        if (idx < 0 || idx >= NN_FORK_MAX_) {
+            errno = EINVAL;
+            break;
+        }
+#ifndef NN_HAVE_WINDOWS
+        nn_fork_strategy = &nn_fork_strategies[idx];
+#endif
+        return 0;
+    default:
+        errno = ENOPROTOOPT;
+        break;
+    }
+    return -1;
+}
+
+int nn_global_lock_all_sockets (void)
+{
+    int i;
+
+    if (self.socks && self.nsocks) {
+        for (i = 0; i != NN_MAX_SOCKETS; ++i)
+            if (self.socks [i])
+                nn_ctx_enter (nn_sock_getctx (self.socks [i]));
+    }
+}
```

Comment:
This might be a multi-threading race; let's check into whether `nn_glock_lock()` needs to be held. Specifically -- in the `if` test, consider if the socket is destroyed in a parallel thread just after the test but before the next line it's guarding.